### PR TITLE
switch to new .thumb method from dor-services

### LINKED ISSUE #32: Update file_id method to use new thumb logic in dor-services gem
use .thumb method in dor-services and escape / in output to %2F

### PR DESCRIPTION
**Pull request by peetucket**
fixes #32

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -63,11 +63,11 @@ GEM
       capistrano-releaseboard
     docile (1.1.5)
     docopt (0.5.0)
-    domain_name (0.5.20160615)
+    domain_name (0.5.20160826)
       unf (>= 0.0.5, < 1.0.0)
     dor-rights-auth (1.2.0)
       nokogiri
-    dor-services (5.10.1)
+    dor-services (5.11.0)
```
Comment (drh-stanford) [inline Gemfile.lock]: there's a `5.12.0` now

File: spec/update_marc_spec.rb [hunk 2]
```diff
@@ -189,236 +187,172 @@
...
-      expect(d).to receive(:id).and_return('bb111bb2222')
-      expect(d).to receive(:datastreams).exactly(4).times.and_return({'contentMetadata' => content_metadata_ds})
-      expect(content_metadata_ds).to receive(:ng_xml).twice.and_return(content_metadata_ng_xml)
+      expect(d).to receive(:id).twice.and_return('bb111bb2222')
+      expect(d).to receive(:datastreams).exactly(3).times.and_return({'contentMetadata' => content_metadata_ds})
+      expect(content_metadata_ds).to receive(:ng_xml).and_return(content_metadata_ng_xml)
 
       updater = Dor::UpdateMarcRecordService.new(d)
-      expect(updater.file_id).to eq('|xfile:bb111bb2222%2Fwt183gy6220_00_0001.jp2')
+      expect(updater.thumb).to eq('|xfile:bb111bb2222%2Fwt183gy6220_00_0001.jp2')
```
Comment (drh-stanford) [inline spec/update_marc_spec.rb]: what is this `|xfile:` business? I couldn't find any documentation on it in the codebase

**Comment by drh-stanford** [comment]
this PR looks good, just a couple minor comments...

File: spec/update_marc_spec.rb (hunk 2 above)
Comment (peetucket) [inline spec/update_marc_spec.rb]: the |xfile: syntax is something specific to the line in the file that is generated for updating the marc record... i am not sure of the specifics either - i presume it is a MARC specification

File: Gemfile.lock (hunk 1 above)
Comment (peetucket) [inline Gemfile.lock]: well, i just tried updating to 5.12.0 and now it is force downgrading activesupport to a lower version of 4, which bundle-audit is complaining about...need some investigation

Comment (drh-stanford) [inline Gemfile.lock]: yes, I can confirm that `dor-services v5.12.0` breaks `bundle-audit` when you do a `bundle update dor-services`:

```
Name: activesupport
Version: 4.0.13
Advisory: CVE-2015-3227
Criticality: Unknown
URL: https://groups.google.com/forum/#!topic/rubyonrails-security/bahr2JLnxvk
Title: Possible Denial of Service attack in Active Support
Solution: upgrade to >= 4.2.2, ~> 4.1.11, ~> 3.2.22
```

so I would suggest:

```
-gem 'dor-services', '~> 5.11'
+gem 'dor-services', '~> 5.11.1' # 5.12.0 break with a bundle-audit error for activesupport
```

Comment (cbeer) [inline Gemfile.lock]: Hm. I don't know why it is downgrading `activesupport` on you. Pinning it in the app seems to be ok, though.

Comment (drh-stanford) [inline Gemfile.lock]: hmmm. it looks like the problem is doing a `bundle update dor-services`... if you just change:

```
-gem 'dor-services', '~> 5.11'
+gem 'dor-services', '~> 5.12'
```

and then do a `bundle install` (not update), `bundle-audit` will pass...

Comment (peetucket) [inline Gemfile.lock]: I did some sleuthing in the Gemfile.lock after upgrading and couldn't find which update was causing activesupport to downgrade.

Comment (cbeer) [inline Gemfile.lock]: e.g. https://github.com/sul-dlss/dor-services-app/pull/34

Comment (peetucket) [inline Gemfile.lock]: what do you think of @drh-stanford comment right above yours @cbeer ?  Should we still pin activesupport just to be sure?

**Comment by peetucket** [comment]
ok, pinned activesupport via #34 and updated to 5.12, bundle-audit passes, bundle install works as expected

**Comment by drh-stanford** [comment]
sorry, i didn't see #34 before my last comment. i looked over the resulting `Gemfile.lock` and it's correctly upgrading to ActiveFedora 6.8.0 and rubydora 1.9.0 (which were both released recently).

**Pull request merged**